bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/api/BValueCreator.java

```diff
@@ -0,0 +1,438 @@
+      * @return byte array
+      */
+     public static BArray createArrayValue(byte[] values) {
+         return new ArrayValue(values);
+     }
+
+     /**
+      * Creates a new float array.
+      *
+      * @param values initial array values
+      * @return float array
+      */
+     public static BArray createArrayValue(double[] values) {
+         return new ArrayValue(values);
+     }
+
+     /**
+      * Creates a new string array.
+      *
+      * @param values initial array values
+      * @return string array
+      */
+     public static BArray createArrayValue(String[] values) {
+         return new ArrayValue(values);
+     }
+
+     /**
+      * Create a new Ref value array.
+      *
+      * @param values initial array values
+      * @param type {@code BType} of the array. Do not use for primitive types.
+      * @return ref Value array
+      */
+     public static BArray createArrayValue(Object[] values, BType type) {
+         return new ArrayValue(values, type);
+     }
+
+     /**
+      * Create a ref value array with given maximum length.
+      *
+      * @param values initial array values
+      * @param type {@code BType} of the array. Do not use for primitive types
+      * @param length maximum length
+      * @return fixed length ref value array
+      */
+     public static BArray createArrayValue(Object[] values, BType type, int length) {
+         return new ArrayValue(type, length);
+     }
+
+     /**
+      * Create a decimal from given value.
+      *
+      * @param value the value of the decimal
+      * @return decimal value
+      */
+     public static BDecimal createDecimalValue(BigDecimal value) {
+         return new DecimalValue(value);
+     }
+
+     /**
+      * Create a decimal from given string.
+      *
+      * @param value string value
+      * @return decimal value
+      */
+     public static BDecimal createDecimalValue(String value) {
+         return new DecimalValue(value);
+     }
+
+     /**
+      * Create a decimal from given string and value kind.
+      *
+      * @param value string value
+      * @param valueKind value kind
+      * @return decimal value
+      */
+     public static BDecimal createDecimalValue(String value, DecimalValueKind valueKind) {
+         return new DecimalValue(value, valueKind);
+     }
+
+     /**
+      * Create error value with given reason and error details.
+      *
+      * @param reason error reason
+      * @param details error detail
+      * @return error value
+      */
+     public static BError createErrorValue(String reason, Object details) {
+         return new ErrorValue(reason, details);
+     }
+
+     /**
+      * Create error value with given type, reason and details.
+      *
+      * @param type {@code BType} of the error
+      * @param reason error reason
+      * @param details error details
+      * @return error value
+      */
+     public static BError createErrorValue(BType type, String reason, Object details) {
```

Comment:
Should the parameter be `BErrorType type` instead of `BType type`?

I think we should do this for all other methods as well, to make sure the type safety.